phygnn/utilities/tf_utilities.py

```diff
@@ -19,3 +20,174 @@ def tf_log10(x):
     num = tf.math.log(x)
     den = tf.math.log(tf.constant(10, dtype=num.dtype))
     return num / den
+
+
+def _mean_fill(x, mask):
+    """Fill NaN values in the input tensor with the mean of the non-NaN.
+    If all values are NaN, fill with 0. This is assumed to be either a 3D
+    or 4D tensor, with the trailing feature channel removed.
+
+    Parameters
+    ----------
+    x : tf.Tensor
+        Tensor with NaNs and shape (n_obs, spatial_1, spatial_2) or
+        (n_obs, spatial_1, spatial_2, n_temporal)
+    mask : tf.Tensor
+        Boolean mask of the non-NaN locations in the original x
+
+    Returns
+    -------
+    x : tf.Tensor
+        Input tensor with NaN values filled with the mean of the non-NaN
+        values or 0 if all values are NaN.
+    mask : tf.Tensor
+        Mask of the input tensor where True is not NaN and False is NaN.
+    """
+    not_nan_float = tf.cast(mask, x.dtype)
+
+    hi_res_zeroed = tf.where(mask, x, tf.zeros_like(x))
+
+    count = tf.reduce_sum(not_nan_float)
+    total = tf.reduce_sum(hi_res_zeroed)
+    mean = tf.math.divide_no_nan(total, count)
```

Comment:
Maybe there's a reason you're doing this interesting mean operation but consider more intuitive syntax based on this example instead:
```
import numpy as np
import tensorflow as tf
arr = np.random.uniform(0, 1, (4, 4))
tensor = tf.convert_to_tensor(arr)
mask = tensor > 0.5
mean = tf.reduce_mean(tensor[mask])
out = tf.where(mask, tensor, mean)

Reply:
Yeah not totally sure why i did it this way lol